Index: dwm/config.def.h
===================================================================
--- dwm.orig/config.def.h
+++ dwm/config.def.h
@@ -23,6 +23,11 @@ static const char *colors[][3]      = {
 };
 
 static const char *const autostart[] = {
+    "lxpolkit", NULL,
+	"picom", NULL,
+    "nitrogen", "--restore", NULL,
+    "/bin/sh", "-c", "/home/moena/.local/share/dwm/autostart.sh", NULL,
+	"slstatus", NULL,
 	NULL /* terminate */
 };
 
